### PR TITLE
Preview serialization update

### PR DESCRIPTION
**Pull request by fengzhou-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/NetworkInterfaceIPConfigurationData.Serialization.cs [hunk 1]
```diff
@@ -130,7 +132,8 @@ internal static NetworkInterfaceIPConfigurationData DeserializeNetworkInterfaceI
             Optional<ETag> etag = default;
             Optional<ResourceIdentifier> id = default;
             Optional<string> name = default;
-            Optional<ResourceType> type = default;
+            Optional<ResourceType?> type = default;
```
Comment (fengzhou-msft) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/NetworkInterfaceIPConfigurationData.Serialization.cs:135]: This looks problematic.

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -97,7 +97,7 @@
     <PackageReference Update="Azure.Security.KeyVault.Certificates" Version="4.2.0" />
     <PackageReference Update="Azure.Storage.Blobs" Version="12.12.0" />
     <PackageReference Update="Azure.Storage.Queues" Version="12.10.0" />
-    <PackageReference Update="Azure.ResourceManager" Version="1.0.0" />
+    <PackageReference Update="Azure.ResourceManager" Version="1.2.0-alpha.20220609.1" />
```
Comment (m-nash) [inline eng/Packages.Data.props:100]: This is fine as a POC but shouldn't be checked in.

File: sdk/resources/Azure.ResourceManager.Resources/src/Generated/Models/AzureCliScript.Serialization.cs [hunk 3]
```diff
@@ -117,7 +117,7 @@ internal static AzureCliScript DeserializeAzureCliScript(JsonElement element)
             ResourceIdentifier id = default;
             string name = default;
             ResourceType type = default;
-            SystemData systemData = default;
+            Optional<SystemData> systemData = default;
```
Comment (m-nash) [inline sdk/resources/Azure.ResourceManager.Resources/src/Generated/Models/AzureCliScript.Serialization.cs:120]: why does this result in only serialization changes without changing any base types in resources package?

File: sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/WritableResourceData.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+#nullable disable
+
+using Azure.Core;
+
+namespace Azure.ResourceManager.Models
+{
+    /// <summary> Common fields that are returned in the response for all Azure Resource Manager resources. </summary>
+    [ReferenceType(new string[]{"SystemData"})]
+    public abstract partial class WritableResourceData
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/WritableResourceData.cs:12]: Did we run the same analysis to see if we want a WriteableTrackedResourceData as well?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: sdk/resources/Azure.ResourceManager.Resources/src/Generated/Models/AzureCliScript.Serialization.cs (hunk 3 above)
Comment (fengzhou-msft) [reply sdk/resources/Azure.ResourceManager.Resources/src/Generated/Models/AzureCliScript.Serialization.cs:120]: Previously, during replacement with the common type, we hardcoded `Required` as true for its properties. I have updated generator to fix that. See SystemObjectType code [here](https://github.com/Azure/autorest.csharp/pull/2303/files#diff-32a4ccc7bb0ed76e7043f8ac39b8b80af8dcbf850aeb1add76ccd922b6215c8e).

File: sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/WritableResourceData.cs (hunk 4 above)
Comment (fengzhou-msft) [reply sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/WritableResourceData.cs:12]: Yes. It turns out to be not common for id, name or type to be writable in TrackedResourceData. The only valid case I met with is writable ids in network.

**Pull request closed without merging**